components/calendar/src/any_calendar.rs

```diff
@@ -245,6 +328,50 @@ impl AnyDateInner {
     }
 }
 
+/// Convenient type for selecting the kind of AnyCalendar to construct
+#[non_exhaustive]
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub enum AnyCalendarKind {
+    Gregorian,
+    Buddhist,
+    Japanese,
+    Indian,
+    Coptic,
+    Iso,
+}
+
+impl AnyCalendarKind {
+    pub fn from_bcp47(x: &str) -> Option<Self> {
+        Some(match x {
+            "gregory" => AnyCalendarKind::Gregorian,
+            "buddhist" => AnyCalendarKind::Buddhist,
+            "indian" => AnyCalendarKind::Indian,
+            "coptic" => AnyCalendarKind::Coptic,
+            "iso" => AnyCalendarKind::Iso,
+            _ => return None,
+        })
+    }
+
+    pub fn as_bcp47(&self) -> &'static str {
+        match *self {
+            AnyCalendarKind::Gregorian => "gregory",
```

Comment:
Nit: These strings also exist elsewhere, in *components/datetime/src/calendar.rs*. It would be nice not to duplicate them in library code.

Also, note that in the datetime component, we use `extensions::unicode::Value` as the canonical type for BCP-47, rather than the string. I'm fine also accepting the string in another function.

Reply:
ok, I still prefer putting the strings in either one place for the other, but if you intend to clean this up, it's fine